### PR TITLE
[Design] [MIS] MIS of User Module

### LINKED ISSUE #364: [Design] [MIS] MIS - Users
## Content

- MIS

## Rubric Sections

- DetDesDoc: Syntax
- DetDesDoc: Semantics
- DetDesDoc: EnoughToBuild
- Important Parts Formalized (LO_SpecMath)

### PR DESCRIPTION
**Pull request by arkinmodi**
Close https://github.com/arkinmodi/project-sayyara/issues/364

### CODE REVIEW

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 1]
```diff
@@ -182,77 +185,193 @@ \section{Module Decomposition}
...
+\begin{table}[H]
+	\begin{tabular}{|l|l|l|}
+		\hline
+		\textbf{Output Name}   & \textbf{Output Type} & \textbf{Description}                \\
+		\hline
+		email                  & String               & Email of Customer                   \\
+		\hline
+		password               & String               & Password of Customer Account        \\
+		\hline
+		first\_name            & String               & First Name of Customer              \\
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex:216]: add phone number

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 2]
```diff
@@ -182,77 +185,193 @@ \section{Module Decomposition}
...
+
+\textbf{CreateEmployeeType}
+
+\begin{table}[H]
+	\begin{tabular}{|l|l|l|}
+		\hline
+		\textbf{Output Name} & \textbf{Output Type} & \textbf{Description}               \\
+		\hline
+		email                & String               & Email of Employee                  \\
+		\hline
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex:243]: add phone number

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 3]
```diff
@@ -182,77 +185,193 @@ \section{Module Decomposition}
...
+\textbf{CreateShopOwnerType}
+
+% TODO: Add Shop Fields
+\begin{table}[H]
+	\begin{tabular}{|l|l|l|}
+		\hline
+		\textbf{Output Name} & \textbf{Output Type} & \textbf{Description}             \\
+		\hline
+		email                & String               & Email of Shop Owner              \\
+		\hline
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex:266]: add phone number

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 4]
```diff
@@ -182,77 +185,193 @@ \section{Module Decomposition}
...
+		password             & String               & Password of Shop Owner's Account \\
+		\hline
+		first\_name          & String               & First Name of Shop Owner         \\
+		\hline
+		last\_name           & String               & Last Name of Shop Owner          \\
+		\hline
+		shop.---             & String               & \dots                            \\
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex:275]: Can we leave a todo comment for the shop fields? So we don't forget about this

Comment (arkinmodi) [reply docs/Design/SoftDetailedDes/MIS.tex:275]: https://github.com/arkinmodi/project-sayyara/blob/528130647314ced77a237ccbac8cba6274b17cb0/docs/Design/SoftDetailedDes/MIS.tex#L255

https://github.com/arkinmodi/project-sayyara/issues/371#issuecomment-1379555771

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 5]
```diff
@@ -182,77 +185,199 @@ \section{Module Decomposition}
...
+		\hline
+		\textbf{Output Name} & \textbf{Output Type} & \textbf{Description} \\
+		\hline
+		id                   & String               & User ID              \\
 		\hline
-		\textbf{Name}    & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+		firstName            & String               & First Name of User   \\
 		\hline
-		\wss{accessProg} & -           & -            & -                   \\
+		lastName             & String               & Last Name of User    \\
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex:291]: should phone number be included in the `AuthorizeReturnType` now?

Comment (arkinmodi) [reply docs/Design/SoftDetailedDes/MIS.tex:291]: I don't think so. Do you need the phone number in the `useSession()` hook? If yes, we can add it.

Comment (LeonSo7) [reply docs/Design/SoftDetailedDes/MIS.tex:291]: Probably not for the time being. We can leave it out.

**Review by LeonSo7 — APPROVED** [review]

**Pull request merged**